### DISCUSSION TITLE
apt-cacher-ng related script

### DISCUSSION THREAD

**Discussion by mattvanmater** [tteck/Proxmox#2474] [Ideas]
I noticed you recently added a apt-cacher-ng LXC to your scripts!  I've been running my own apt-cacher-ng for a long time now and I thought I would share a script that i've found useful.  If the apt-cacher proxy is not available for any reason, any LXC or VM that depends upon it will hang when trying to perform apt-update through the apt proxy.  This script will check to see if the proxy is available prior to trying to use it, and if not it will directly communicate with your apt repos.

The script:
`echo "Acquire::http::Proxy-Auto-Detect "/usr/local/bin/apt-proxy-detect.sh";" >> /etc/apt/apt.conf.d/00aptproxy

cat > /usr/local/bin/apt-proxy-detect.sh <<'EOF'
#!/bin/bash
IP=---your DNS name or ip here---
PORT=3142
if nc -w1 -z $IP $PORT; then
    echo -n "http://${IP}:${PORT}"
else
    echo -n "DIRECT"
fi
EOF

chmod +x /usr/local/bin/apt-proxy-detect.sh`

Perhaps you can re-use it somewhere (install.func, build.func, automatically inject it into containers and VMs?).  The script above is the first thing I run after building a new Debian based LXC or VM.  

Side note: I also have it running on my Linux based laptops.  It is particularly useful / important if i take my laptop away from my home and I need to update/install software.  I realize it's not applicable for most proxmox environments, but it's handy to have and be consistent across all hosts both physical and virtual.

**Comment by tteck** [comment]
That sounds like a fantastic idea! I definitely plan to include it in the scripts.

**Comment by tteck** [comment]
It has been added. https://github.com/tteck/Proxmox/commit/ffd2fc3b868ab802e14f63cd49e0659347c33461

**Comment by tteck** [comment]
For anyone wanting to switch to this method, change `<apt-cacher-ng-IP>` to your proxy IP address, then execute the snippet within the LXC console.
```
echo "Acquire::http::Proxy-Auto-Detect \"/usr/local/bin/apt-proxy-detect.sh\";" >/etc/apt/apt.conf.d/00aptproxy
cat <<EOF >/usr/local/bin/apt-proxy-detect.sh
#!/bin/bash
if nc -w1 -z <apt-cacher-ng-IP> 3142; then
echo -n "http://<apt-cacher-ng-IP>:3142"
else
echo -n "DIRECT"
fi
EOF
chmod +x /usr/local/bin/apt-proxy-detect.sh
```
It will overwrite your previous `/etc/apt/apt.conf.d/00aptproxy`
